packages/relay/src/lib/clients/sdkClient.ts

```diff
@@ -643,60 +644,142 @@ export class SDKClient {
+
+        this.captureMetrics(
+          SDKClient.transactionMode,
+          fileAppendTx.constructor.name,
+          Status.Success,
+          await this.calculateFileAppendTxTotalTinybarsCost(fileAppendTx),
+          0,
+          callerName,
+          interactingEntity,
+        );
+      }
+
+      // Ensure that the calldata file is not empty
+      if (fileId) {
+        const fileSize = await (await new FileInfoQuery().setFileId(fileId).execute(client)).size;
 
-      if (callData.length > 0 && fileSize.isZero()) {
-        throw new SDKClientError({}, `${requestIdPrefix} Created file is empty. `);
+        if (callData.length > 0 && fileSize.isZero()) {
+          throw new SDKClientError({}, `${requestIdPrefix} Created file is empty. `);
+        }
+        this.logger.trace(`${requestIdPrefix} Created file with fileId: ${fileId} and file size ${fileSize}`);
       }
-      this.logger.trace(`${requestIdPrefix} Created file with fileId: ${fileId} and file size ${fileSize}`);
+    } catch (error: any) {
+      const sdkClientError = new SDKClientError(error, error.message);
+      let transactionFee: number | Hbar = 0;
+
+      // if valid network error utilize transaction id
+      if (sdkClientError.isValidNetworkError()) {
+        try {
+          const transactionCreateRecord = await new TransactionRecordQuery()
+            .setTransactionId(fileCreateTx.transactionId!)
+            .setNodeAccountIds(fileCreateTx.nodeAccountIds!)
+            .setValidateReceiptStatus(false)
+            .execute(this.clientMain);
+          transactionFee = transactionCreateRecord.transactionFee;
+          this.hbarLimiter.addExpense(transactionFee.toTinybars().toNumber(), currentDateNow);
+
+          this.captureMetrics(
+            SDKClient.transactionMode,
+            fileCreateTx.constructor.name,
+            sdkClientError.status,
+            transactionFee.toTinybars().toNumber(),
+            transactionCreateRecord?.contractFunctionResult?.gasUsed,
+            callerName,
+            interactingEntity,
+          );
+
+          const transactionAppendRecord = await new TransactionRecordQuery()
+            .setTransactionId(fileAppendTx.transactionId!)
+            .setNodeAccountIds(fileAppendTx.nodeAccountIds!)
+            .setValidateReceiptStatus(false)
+            .execute(this.clientMain);
+          transactionFee = transactionAppendRecord.transactionFee;
+          this.hbarLimiter.addExpense(transactionFee.toTinybars().toNumber(), currentDateNow);
+
+          this.captureMetrics(
+            SDKClient.transactionMode,
+            fileCreateTx.constructor.name,
+            sdkClientError.status,
+            transactionFee.toTinybars().toNumber(),
+            transactionCreateRecord?.contractFunctionResult?.gasUsed,
+            callerName,
+            interactingEntity,
+          );
+
+          this.logger.info(
```

Comment:
Add logic to log for both FileCreate and `FileAppend` separately if applicable

Reply:
Done.